lambdas/handlers/authoriser_handler.py

```diff
@@ -0,0 +1,221 @@
+"""
+This code has been modified from AWS blueprint. Below is the original license:
+
+Copyright 2015-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License. A copy of the License is located at
+     http://aws.amazon.com/apache2.0/
+or in the "license" file accompanying this file.
+This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and limitations under the License.
+"""
+
+import logging
+import os
+import re
+
+import boto3
+import botocore
+import jwt
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def lambda_handler(event, context):
+    user = "Unauthorised"
+    ssm_public_key_parameter_name = os.environ["SSM_PARAM_JWT_TOKEN_PUBLIC_KEY"]
+
+    try:
+        client = boto3.client("ssm", region_name="eu-west-2")
+        ssm_response = client.get_parameter(
+            Name=ssm_public_key_parameter_name, WithDecryption=True
+        )
+        public_key = ssm_response["Parameter"]["Value"]
+
+        # For now we just assume the user authenticate with a signed token in this form:
+        # { "user_role": "Unauthorised"  or  "GP"  or  "PCSE" }
+        # The token is going to be issued by the lambda of ticket PRMDR-167
+
+        decoded = jwt.decode(
+            event["authorizationToken"], public_key, algorithms=["RS256"]
+        )
+        user = decoded["user_role"]
+        logger.info(f"decoded JWT: {decoded}")
+    except botocore.exceptions.ClientError as e:
+        logger.error(e)
+    except jwt.PyJWTError as e:
+        logger.error(f"error while decoding JWT: {e}")
+    except KeyError as e:
+        logger.error(e)
+
+    principalId = ""
+    tmp = event["methodArn"].split(":")
```

Comment:
can we rename this to something more meaningful?

Reply:
Thanks!  Addressed in this commit: https://github.com/nhsconnect/national-document-repository/pull/42/commits/c0192931217fdc6f88a751322e92b745595cfa75